src/cli/index.js

```diff
@@ -398,14 +399,32 @@ export function main({
     });
 }
 
-export default function start() {
-  // ignore all arguments after a --
-  const doubleDashIndex = process.argv.findIndex(element => element === '--');
-  const startArgs = process.argv.slice(0, 2);
-  const args = process.argv.slice(2, doubleDashIndex === -1 ? process.argv.length : doubleDashIndex);
-  const endArgs = doubleDashIndex === -1 ? [] : process.argv.slice(doubleDashIndex + 1, process.argv.length);
+async function start(): Promise<void> {
+  const rc = getRcConfigForCwd(process.cwd());
+  const yarnPath = rc['yarn-path'];
 
-  main({startArgs, args, endArgs});
+  if (yarnPath && process.env.YARN_IGNORE_PATH !== '1') {
```

Comment:
Maybe we can check if the current yarn path matching `yarnPath` too to get rid of the environment variable? I mean I'm fine with the env variable since it also allows us to bypass this behavior. Just asking what you think :)